django_codemod/visitors/urls.py

```diff
@@ -99,4 +99,11 @@ def parse_next_group(self, left_to_parse):
     def check_route(self, route):
         """Check that route doesn't contain anymore regex."""
         if route != re.escape(route):
-            raise PatternNotSupported("Route contains regex")
+            # re.escape can give a false positive for dashes,
+            # which are common in URLs (it always escapes them).
+            # If no _other_ regex metacharacter exists in the route,
+            # it's probably not a regex after all.
```

Comment:
I've been thinking about this case since I wrote the code and I think I agree 😄 We should probably not use the `re.escape` from Python 3.7+, but use our own, without dashes.

Given that it would be specific to this module, I think the real fix would be to:

- Move our `re.escape` function into this module and call it something more appropriate (e.g. `escape_patterns_in_route`, but I'm open to suggestions).
- Remove the `-` from the `_special_chars_map`

This way, there is no need to do the extra separate check here, we can keep do a single check `route != escape_patterns_in_route(route)` or equivalent.

Reply:
Or... What if we just skip the `escape()` check altogether and do

```python
re_special_chars = set("()[]{}?*+-|^$\\.&~# \t\n\r\v\f")
re_non_dash_special_chars = re_special_chars - {"-"}

# ...

if set(route) & re_non_dash_special_chars:
    raise PatternNotSupported(f"Route {route} contains special characters that may be part of a regexp")
```

Reply:
Agree that the check can be simplified. I was being conservative in the initial implementation by reusing the same `re.escape` implementation as Python.

I'm wondering about the set difference, in your last suggestion though. Why not define a single special characters set without `-`?

Reply:
Can do that - I was just thinking there might be an use for a set of all likely regexp special characters :)

Reply:
Rewrote the patch to use simpler set check. :)